jobclient/java/src/main/java/com/twosigma/cook/jobclient/Job.java

```diff
@@ -1264,7 +1268,9 @@ public static Job parseFromJSON(JSONObject json, InstanceDecorator decorator){
                 }
             }
         }
-        jobBuilder.addInstances(Instance.parseFromJSON(json.getJSONArray("instances"), decorator));
+        if (json.has("instances")) {
```

Comment:
I'm not sure why this is required as a field.

Reply:
I'm not following, where is it required?

Reply:
Oh if you have a JSON blob that doesn't populate "instances", this line will fail. In fact there's even a hacky workaround in the tests to [convert a Job to JSON](https://github.com/twosigma/Cook/blob/master/jobclient/java/src/test/java/com/twosigma/cook/jobclient/JobTest.java#L82).

Reply:
Basically if you have a JSON payload and you call `parseFromJSON`, you won't be able to get a Job object unless the JSON has defined specific fields, e.g. "uuid", "mem", "cpus", just to name a few.

Reply:
I edited this to be able to parse from JSON -> Builder instead. This API is much more intuitive for users who need to go from JSON -> Job, but may need to make edits in the middle to the Job.

In our use case we have a map of parameters, which we want to merge with another map of parameters. To accomplish that without this CR we would have to do JSON -> Job -> Builder -> Job.